packages/ibm-products/src/components/TagOverflow/TagOverflow.js

```diff
@@ -0,0 +1,266 @@
+      if (id && node && map.get(id) !== node.offsetWidth) {
+        map.set(id, node.offsetWidth);
+      }
+    };
+
+    const getVisibleItems = useCallback(() => {
+      if (!itemRefs.current) {
+        return items;
+      }
+
+      const map = getMap();
+      const overflowContanerWidth =
+        overflowRef.current.offsetWidth > overflowIndicatorWidth
+          ? overflowRef.current.offsetWidth
+          : overflowIndicatorWidth;
+      const maxWidth = containerWidth - overflowContanerWidth;
+
+      let childrenWidth = 0;
+      let maxReached = false;
+
+      return items.reduce((prev, cur) => {
+        if (!maxReached) {
+          const itemWidth = map.get(cur.id) + measurementOffset;
+          const fits = itemWidth + childrenWidth < maxWidth;
+
+          if (fits) {
+            childrenWidth += itemWidth;
+            prev.push(cur);
+          } else {
+            maxReached = true;
+          }
+        }
+        return prev;
+      }, []);
+    }, [itemRefs, overflowRef, containerWidth, items]);
+
+    const ItemComponent = ({ templateProps }) => {
+      const itemRef = (node) => itemRefHandler(templateProps.id, node);
+      let className = templateProps.className
+        ? `${blockClass}__item ${templateProps.className}`
+        : `${blockClass}__item`;
+
+      const ItemComp = () =>
+        React.createElement(itemTemplate, {
+          ref: itemRef,
+          ...templateProps,
+          className,
+        });
+
+      return <ItemComp></ItemComp>;
+    };
+
+    ItemComponent.propTypes = {
+      templateProps: PropTypes.shape({
+        id: PropTypes.string.isRequired,
+        className: PropTypes.string,
+      }).isRequired,
+    };
+
+    useEffect(() => {
+      let visibleItemsArr = getVisibleItems();
+
+      if (maxVisible && maxVisible < visibleItemsArr.length) {
+        visibleItemsArr = visibleItemsArr?.slice(0, maxVisible);
+      }
+
+      const hiddenItems = items?.slice(visibleItemsArr.length);
+      const overflowItemsArr = hiddenItems?.map((item) => {
+        return { type: tagType, label: item.label };
+      });
+
+      setVisibleItems(visibleItemsArr);
+      setOverflowItems(overflowItemsArr);
+    }, [containerWidth, items, maxVisible, getVisibleItems]);
+
+    return (
+      <div
+        {
+          // Pass through any other property values as HTML attributes.
+          ...rest
+        }
+        className={cx(
+          blockClass, // Apply the block class to the main HTML element
+          className, // Apply any supplied class names to the main HTML element.
+          // example: `${blockClass}__template-string-class-${kind}-n-${size}`,
+          {
+            // switched classes dependant on props or state
+            // example: [`${blockClass}__here-if-small`]: size === 'sm',
+          }
+        )}
+        ref={containerRef}
+        role="main"
+        {...getDevtoolsProps(componentName)}
+      >
+        {!!visibleItems?.length &&
```

Comment:
since this variable is initiated with an empty array i think you can simplify this to just `visibleItems.length > 0`